x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -575,6 +587,10 @@ protected void doSaveState(IndexerState indexerState, Map<String, Object> positi
                             DataFrameTransformCheckpointingInfo.EMPTY), // TODO should this be null
                     ActionListener.wrap(
                             r -> {
+                                // for auto stop shutdown the task
+                                if (state.getTaskState().equals(DataFrameTransformTaskState.STOPPED)) {
+                                    onStop();
+                                }
```

Comment:
I am unsure whether next.run() should be called 1st or 2nd. Right now it does not matter as it's an empty lambda for this case